tests/drum/test_fit_inference.py

```diff
@@ -0,0 +1,480 @@
+        # TODO: check for graceful failure for these cases
+        if (framework == SKLEARN_ANOMALY and problem != ANOMALY) or (
+            problem == ANOMALY and framework != SKLEARN_ANOMALY
+        ):
+            return
+
+        custom_model_dir = _create_custom_model_dir(
+            res,
+            tmp_path,
+            framework,
+            problem,
+            language,
+            is_training=True,
+            nested=nested if language == PYTHON else False,  # TODO: support nested R files
+        )
+
+        input_dataset = res.datasets(framework, problem)
+
+        weights_cmd, input_dataset, __keep_this_around = self._add_weights_cmd(
+            weights, input_dataset
+        )
+
+        output = tmp_path / "output"
+        output.mkdir()
+
+        cmd = "{} fit --code-dir {} --input {} --verbose ".format(
+            ArgumentsOptions.MAIN_COMMAND, custom_model_dir, input_dataset
+        )
+        if problem == ANOMALY:
+            cmd += " --unsupervised"
+        else:
+            cmd += " --target {}".format(res.targets(problem))
+
+        if use_output:
+            cmd += " --output {}".format(output)
+        if problem == BINARY:
+            cmd = _cmd_add_class_labels(cmd, res.class_labels(framework, problem))
+        if docker:
+            cmd += " --docker {} ".format(docker)
+
+        cmd += weights_cmd
+
+        _exec_shell_cmd(
+            cmd, "Failed in {} command line! {}".format(ArgumentsOptions.MAIN_COMMAND, cmd)
+        )
+
+    def _create_fit_input_data_dir(
+        self, get_target, get_dataset_filename, input_dir, problem, weights
+    ):
+        input_dir.mkdir(parents=True, exist_ok=True)
+
+        input_dataset = get_dataset_filename(None, problem)
+        df = pd.read_csv(input_dataset)
+
+        # Training data
+        with open(os.path.join(input_dir, "X.csv"), "w+") as fp:
+            if problem == ANOMALY:
+                feature_df = df
+            else:
+                feature_df = df.loc[:, df.columns != get_target(problem)]
+            feature_df.to_csv(fp, index=False)
+
+        if problem != ANOMALY:
+            # Target data
+            with open(os.path.join(input_dir, "y.csv"), "w+") as fp:
+                target_series = df[get_target(problem)]
+                target_series.to_csv(fp, index=False, header="Target")
+
+        # Weights data
+        if weights:
+            df = pd.read_csv(input_dataset)
+            weights_data = pd.Series(np.random.randint(1, 3, len(df)))
+            with open(os.path.join(input_dir, "weights.csv"), "w+") as fp:
+                weights_data.to_csv(fp, header=False)
+
+    @pytest.mark.parametrize("framework", [SKLEARN, XGB, KERAS, SKLEARN_ANOMALY])
+    @pytest.mark.parametrize("problem", [BINARY, REGRESSION, ANOMALY])
+    @pytest.mark.parametrize("language", [PYTHON])
+    @pytest.mark.parametrize("weights", [WEIGHTS_CSV, None])
+    def test_fit_sh(
```

Comment:
would you mind splitting between fit and inference?

Reply:
Leaving fit and inference in one file as I think fit/inference tests should be split by framework in pairs.